eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,175 @@
+import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/utils';
+
+const rule: TSESLint.RuleModule<'animatedStyle' | 'sharedValue'> = {
+  create: function (context) {
+    return {
+      JSXOpeningElement(node: TSESTree.JSXOpeningElement) {
+        if (node.name.type === AST_NODE_TYPES.JSXMemberExpression) return; // Property-like namespace syntax <Animated.View>
+        if (node.name.type === AST_NODE_TYPES.JSXNamespacedName) return; // XML-based namespace syntax: <Animated:View>
+
+        const sourceCode = context.getSourceCode();
+        const tokensBefore = sourceCode.getTokensBefore(node);
+        const componentName = node?.name?.name;
+
+        main();
+        function main() {
+          if (
+            isVariableDefinedAs(componentName, 'Animated') || //TODO allow to customize rule and provide custom import name
```

Comment:
This seems to also be problematic. One of many common cases here is accidentaly using `Animated` from React Native instead of Reanimated. Therefore, the plugin would miss a lot of its functionality here.

Reply:
Yes, you're right. We could parse imports to detect actual import name. However this improvement could be done in the future, this PR is already big.